### PR TITLE
Jedis try blocks

### PR DESCRIPTION
**Pull request by bbpennel**

### CODE REVIEW

File: metadata/src/main/java/edu/unc/lib/dl/util/DepositStatusFactory.java [hunk 1]
```diff
@@ -97,95 +112,91 @@ public void set(String depositUUID, DepositField field, String value) {
...
-		Jedis jedis = getJedisPool().getResource();
-		String state = jedis.hget(DEPOSIT_STATUS_PREFIX+depositUUID, DepositField.state.name());
-		try {
+		
+		try (Jedis jedis = getJedisPool().getResource()) {
+			String state = jedis.hget(DEPOSIT_STATUS_PREFIX + depositUUID, DepositField.state.name());
 			result = DepositState.valueOf(state);
-		} catch(NullPointerException e) {
-		} catch(IllegalArgumentException e) {
+		} catch (IllegalArgumentException | NullPointerException e) {
```
Comment (harringj) [inline metadata/src/main/java/edu/unc/lib/dl/util/DepositStatusFactory.java]: I know it's how the code was before your changes, but do we really want to be catching a NPE here? Seems like testing for null and then having a path where the code takes appropriate action might be better.

File: metadata/src/main/java/edu/unc/lib/dl/util/JobStatusFactory.java [hunk 2]
```diff
@@ -79,64 +79,59 @@ public void clearStale(String depositUUID) {
...
 	public void completed(String jobUUID) {
-		Jedis jedis = getJedisPool().getResource();
-		jedis.hset(JOB_STATUS_PREFIX + jobUUID,
-				JobField.status.name(), JobStatus.completed.name());
-		jedis.hset(JOB_STATUS_PREFIX + jobUUID,
-				JobField.endtime.name(),
-				String.valueOf(System.currentTimeMillis()));
-		getJedisPool().returnResource(jedis);
+		try (Jedis jedis = getJedisPool().getResource()) {
+			jedis.hset(JOB_STATUS_PREFIX + jobUUID,
```
Comment (harringj) [inline metadata/src/main/java/edu/unc/lib/dl/util/JobStatusFactory.java]: Seems like there might be some extra indents here

File: metadata/src/main/java/edu/unc/lib/dl/util/JobStatusFactory.java [hunk 3]
```diff
@@ -79,64 +79,59 @@ public void clearStale(String depositUUID) {
...
 
 	public void killed(String jobUUID) {
-		Jedis jedis = getJedisPool().getResource();
-		jedis.hset(JOB_STATUS_PREFIX + jobUUID,
-				JobField.status.name(), JobStatus.killed.name());
-		jedis.hset(JOB_STATUS_PREFIX + jobUUID,
-				JobField.endtime.name(),
-				String.valueOf(System.currentTimeMillis()));
-		getJedisPool().returnResource(jedis);
+		try (Jedis jedis = getJedisPool().getResource()) {
```
Comment (harringj) [inline metadata/src/main/java/edu/unc/lib/dl/util/JobStatusFactory.java]: Ditto here

File: metadata/src/main/java/edu/unc/lib/dl/util/JobStatusFactory.java [hunk 4]
```diff
@@ -79,64 +79,59 @@ public void clearStale(String depositUUID) {
...
+					String.valueOf(System.currentTimeMillis()));
+		}
 	}
 
 	public void incrCompletion(String jobUUID, int amount) {
-		Jedis jedis = getJedisPool().getResource();
-		jedis.hincrBy(JOB_STATUS_PREFIX + jobUUID,
-				JobField.num.name(), amount);
-		getJedisPool().returnResource(jedis);
+		try (Jedis jedis = getJedisPool().getResource()) {
```
Comment (harringj) [inline metadata/src/main/java/edu/unc/lib/dl/util/JobStatusFactory.java]: And here

File: metadata/src/main/java/edu/unc/lib/dl/util/JobStatusFactory.java [hunk 5]
```diff
@@ -198,18 +186,14 @@ public String getJobByStatus(String depositUUID, JobStatus status) {
...
 
 		return null;
 	}
 
 	public List<String> getJobsByStatus(String depositUUID, JobStatus status) {
-		Jedis jedis = getJedisPool().getResource();
-
-		List<String> result = new ArrayList<String>();
-		try {
+		List<String> result = new ArrayList<>();
```
Comment (harringj) [inline metadata/src/main/java/edu/unc/lib/dl/util/JobStatusFactory.java]: Just curious about why you reverted to using the raw ArrayList type here. I'm sure there's a good reason, :-)

File: metadata/src/main/java/edu/unc/lib/dl/util/DepositStatusFactory.java (hunk 1 above)
Comment (bbpennel) [inline metadata/src/main/java/edu/unc/lib/dl/util/DepositStatusFactory.java]: Yeah both of these are rather weird cases to be catching, and we aren't generally handling null cases this way in the rest of the code base, so i'll switch that.  I'm not really sure how the IllegalArgumentExceptions would have occurred.

File: metadata/src/main/java/edu/unc/lib/dl/util/JobStatusFactory.java (hunk 2 above)
Comment (bbpennel) [inline metadata/src/main/java/edu/unc/lib/dl/util/JobStatusFactory.java]: This code was actually indented too far before, I shifted it back to the left by one indent.

Comment (bbpennel) [inline metadata/src/main/java/edu/unc/lib/dl/util/JobStatusFactory.java]: Actually sorry I was thinking of DepositStatusFactory.addManifest.  I think the indent is correct here though

File: metadata/src/main/java/edu/unc/lib/dl/util/JobStatusFactory.java (hunk 5 above)
Comment (bbpennel) [inline metadata/src/main/java/edu/unc/lib/dl/util/JobStatusFactory.java]: Because we hates them.

**Pull request merged**